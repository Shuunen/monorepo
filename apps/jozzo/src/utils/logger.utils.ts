import { Logger } from '@shuunen/utils'

export const logger = new Logger()

logger.info('Unique mark : {{ unique-mark }}')
