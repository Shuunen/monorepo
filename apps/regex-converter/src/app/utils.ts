import type { Rule } from './types'

export function applyRules(text: string, rules: Rule[]) {
  let result = text
  for (const rule of rules) {
    if (!rule.enabled) continue
    try {
      result = result.replace(new RegExp(rule.pattern, 'g'), rule.replacement)
    } catch {
      // ignore invalid regex
    }
  }
  return result
}
