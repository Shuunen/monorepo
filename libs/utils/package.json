{
  "dependencies": {
    "esbuild": "^0.25.5",
    "tiny-glob": "^0.2.9"
  },
  "exports": {
    ".": "./src/index.ts",
    "./package.json": "./package.json"
  },
  "name": "@shuunen/utils",
  "nx": {
    "name": "utils",
    "sourceRoot": "libs/utils/src"
  },
  "private": true,
  "type": "module",
  "version": "10.6.0"
}