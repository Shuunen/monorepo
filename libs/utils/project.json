{
  "name": "utils",
  "targets": {
    "list": {
      "cache": true,
      "executor": "nx:run-commands",
      "options": {
        "command": "bun bin/barrel-maker.cli.ts --target=\"./lib/*.ts\" --avoid=\".test.ts\" --ext=\".js\"",
        "cwd": "libs/utils/src"
      },
      "outputs": ["{projectRoot}/src/index.ts"]
    },
    "typecheck": {
      "outputs": ["{projectRoot}/out-tsc/vitest/**/*.d.ts", "{projectRoot}/out-tsc/vitest/tsconfig.tsbuildinfo"]
    }
  }
}
