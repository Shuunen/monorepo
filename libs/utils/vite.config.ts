import { defineConfig } from 'vite'

export default defineConfig(() => ({
  cacheDir: '../../node_modules/.vite/libs/utils',
  plugins: [],
  root: __dirname,
  test: {
    coverage: {
      exclude: ['src/index.ts', 'dist/**', '**/*.config.ts', '**/*.d.ts'],
      provider: 'v8' as const,
      reportsDirectory: './test-output/vitest/coverage',
    },
    environment: 'node',
    globals: true,
    include: ['src/**/*.test.ts'],
    reporters: ['default'],
    watch: false,
  },
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
}))
