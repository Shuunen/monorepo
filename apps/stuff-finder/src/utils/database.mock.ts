import { functionReturningVoid, nbDaysInWeek, sleep } from '@shuunen/utils'
import type { Models } from 'appwrite'
import { vi } from 'vitest'
import type { ItemModel } from '../types/item.types'
import { mockItemModel } from './mock.utils'

export function mockFile(data: Partial<Models.File> = {}) {
  return {
    $createdAt: '2020-03-01T00:00:00.000Z',
    $id: 'some-image-file-uuid-xyz',
    $permissions: [],
    $updatedAt: '2021-08-01T00:00:00.000Z',
    bucketId: 'bucketA',
    chunksTotal: 1,
    chunksUploaded: 1,
    mimeType: 'image/jpeg',
    name: 'fileName-a.jpg',
    signature: 'signature-a',
    sizeOriginal: 123_456,
    ...data,
  } satisfies Models.File as Models.File
}

const createFile = vi.fn(async (bucketId: string, fileId: string, _file: File) => {
  await sleep(nbDaysInWeek)
  return { $id: fileId, bucketId, isThisMockedDataFromMock: true }
})

const deleteFile = vi.fn(async (bucketId: string, fileId: string, _file: File) => {
  await sleep(nbDaysInWeek)
  return { $id: fileId, bucketId, isThisMockedDataFromMock: true }
})

const listFiles = vi.fn(async (_bucketId: string, _queries: [{ limit: number }, { offset: number }]) => {
  await sleep(nbDaysInWeek)
  return { files: [], total: 0 } satisfies Models.FileList as Models.FileList
})

// oxlint-disable-next-line max-params
const createDocument = vi.fn(async (databaseId: string, collectionId: string, documentId: string, data: object) => {
  await sleep(nbDaysInWeek)
  const item = mockItemModel({ $collectionId: collectionId, $databaseId: databaseId, $id: documentId, ...data })
  return item satisfies Models.Document as Models.Document
})

const deleteDocument = vi.fn(async (databaseId: string, collectionId: string, documentId: string) => {
  await sleep(nbDaysInWeek)
  return { $id: documentId, collectionId, databaseId, isThisMockedDataFromMock: true }
})

const listDocuments = vi.fn(async (_databaseId: string, _collectionId: string) => {
  await sleep(nbDaysInWeek)
  return { documents: [], total: 0 } satisfies Models.DocumentList<ItemModel> as Models.DocumentList<ItemModel>
})

// oxlint-disable-next-line max-params
const updateDocument = vi.fn(async (databaseId: string, collectionId: string, documentId: string, data: object) => {
  await sleep(nbDaysInWeek)
  const item = mockItemModel({ $collectionId: collectionId, $databaseId: databaseId, $id: documentId, ...data })
  return item satisfies Models.Document as Models.Document
})

class Databases {
  createDocument = createDocument
  deleteDocument = deleteDocument
  listDocuments = listDocuments
  updateDocument = updateDocument
  constructor(client?: Client) {
    if (client) functionReturningVoid()
  }
}

// oxlint-disable-next-line max-classes-per-file
class Client {
  constructor() {
    functionReturningVoid()
  }
  setProject(project: string) {
    if (project) functionReturningVoid()
    return this
  }
}

class Storage {
  createFile = createFile
  deleteFile = deleteFile
  listFiles = listFiles
  constructor(client?: Client) {
    if (client) functionReturningVoid()
  }
}

const Query = {
  limit: vi.fn((nb: number) => ({ isThisMockedDataFromMock: true, limit: nb })),
  offset: vi.fn((nb: number) => ({ isThisMockedDataFromMock: true, offset: nb })),
}

function reset() {
  createDocument.mockClear()
  createFile.mockClear()
  listFiles.mockClear()
  deleteDocument.mockClear()
  deleteFile.mockClear()
  listDocuments.mockClear()
  updateDocument.mockClear()
  Query.limit.mockClear()
  Query.offset.mockClear()
}

// oxlint-disable-next-line sort-keys
export const databaseMock = {
  // biome-ignore lint/style/useNamingConvention: I cant change this, it's part of the Appwrite SDK
  appwrite: { Client, Databases, Query, Storage },
  createDocument,
  createFile,
  deleteDocument,
  deleteFile,
  listDocuments,
  listFiles,
  // biome-ignore lint/style/useNamingConvention: I cant change this, it's part of the Appwrite SDK
  Query,
  reset,
  updateDocument,
}
