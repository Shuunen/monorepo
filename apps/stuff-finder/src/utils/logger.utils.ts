import { Logger } from '@shuunen/utils'

export const logger = new Logger()
