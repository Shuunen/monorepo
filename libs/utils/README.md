# Utils

> Collection of useful functions to be reused across projects

## Use

```js
import { getTimestamp } from 'utils'
// or for older js : var getTimestamp = require("utils").getTimestamp

console.log('Hello world, current timestamp is', getTimestamp())
```

## Running unit tests

Run `nx test utils` to execute the unit tests via [Vitest](https://vitest.dev/).

## TODO

- [ ] use zod
- [ ] add willOutputToFile option to the logger
- [ ] re-enable useTopLevelRegex in biome

## Thanks

- [Angus Croll](https://github.com/angus-c/just) : for the inspiration of some dependency free functions
- [Box icon](https://www.iconfinder.com/icons/2123914/app_box_essential_ui_icon) : nice icon
- [Boxy Svg](https://boxy-svg.com) : simple & effective svg editor
- [Bun](https://bun.sh) : super fast runtime for JavaScript and TypeScript
- [Esbuild](https://github.com/evanw/esbuild) : an extremely fast JavaScript bundler and minifier
- [Eslint](https://eslint.org) : super tool to find & fix problems
- [Github](https://github.com) : for all their great work year after year, pushing OSS forward
- [Jakub Juszczak's article](https://hackernoon.com/how-to-publish-your-package-on-npm-7fc1f5aae600): nice article about publishing on npm
- [Josh Crowther's repo](https://github.com/jshcrowthe/howto-browser-modules): great ressource about modules
- [Knip](https://github.com/webpro/knip) : super tool to find & fix problems
- [Repo-checker](https://github.com/Shuunen/repo-checker) : eslint cover /src code and this tool the rest ^^
- [Shields.io](https://shields.io) : for the nice badges on top of this readme
- [Svg Omg](https://jakearchibald.github.io/svgomg/) : the great king of svg file size reduction
- [Tsup](https://github.com/egoist/tsup) : super fast js/ts bundler with no config, powered by esbuild <3
- [Watchlist](https://github.com/lukeed/watchlist) : recursively watch a list of directories & run a command on any file system
- [Yoctocolors](https://github.com/sindresorhus/yoctocolors) : for the extra-lightweight color utilities

## Contributors ‚ú®

Thanks goes to these wonderful people :

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
  <tbody>
    <tr>
      <td align="center" valign="top" width="14.28%"><a href="https://fr.linkedin.com/in/romain-racamier"><img src="https://avatars.githubusercontent.com/u/439158?v=4?s=100" width="100px;" alt="Romain Racamier"/><br /><sub><b>Romain Racamier</b></sub></a><br /><a href="#infra-Shuunen" title="Infrastructure (Hosting, Build-Tools, etc)">üöá</a> <a href="https://github.com/Shuunen/utils/commits?author=Shuunen" title="Tests">‚ö†Ô∏è</a> <a href="https://github.com/Shuunen/utils/commits?author=Shuunen" title="Code">üíª</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/babforlife"><img src="https://avatars.githubusercontent.com/u/55501953?v=4?s=100" width="100px;" alt="babforlife"/><br /><sub><b>babforlife</b></sub></a><br /><a href="https://github.com/Shuunen/utils/commits?author=babforlife" title="Code">üíª</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/jordan-boyer"><img src="https://avatars.githubusercontent.com/u/6780830?v=4?s=100" width="100px;" alt="jordan boyer"/><br /><sub><b>jordan boyer</b></sub></a><br /><a href="https://github.com/Shuunen/utils/commits?author=jordan-boyer" title="Code">üíª</a></td>
    </tr>
  </tbody>
</table>

<!-- markdownlint-restore -->
<!-- prettier-ignore-end -->

<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!

## Stargazers over time

[![Stargazers over time](https://starchart.cc/Shuunen/utils.svg?variant=adaptive)](https://starchart.cc/Shuunen/utils)
