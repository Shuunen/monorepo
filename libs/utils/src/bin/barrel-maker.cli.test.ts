import { alignForSnap, Result } from '@monorepo/utils'
import { beforeEach, describe, expect, it, vi } from 'vitest'

const mockFiles = ['atoms/Button.tsx', 'atoms/types.ts', 'atoms/types.d.ts', 'icons/Icon.tsx', 'molecules/Card.tsx', 'molecules/Card.test.tsx', 'molecules/Card.stories.tsx']

describe('barrel-maker cli', () => {
  beforeEach(() => {
    vi.resetModules()
    vi.doMock('tiny-glob', () => ({ default: vi.fn().mockResolvedValue(mockFiles) }))
    vi.doMock('node:fs', () => ({ writeFileSync: vi.fn() }))
  })

  it('make A ext empty and default index.ts', async () => {
    // @ts-expect-error import issue
    const { make } = await import('./barrel-maker.cli')
    const { value } = await make({ ext: '', target: './{atoms,icons,molecules}/*.tsx' })
    expect(value.content).toMatchSnapshot()
    expect(value.files).toMatchSnapshot()
    expect(alignForSnap(value.out)).toContain(`libs/utils/index.ts`)
  })

  it('make B index.js and ext .js', async () => {
    // @ts-expect-error import issue
    const { make } = await import('./barrel-maker.cli')
    const { value } = await make({ ext: '.js', index: 'index.js', target: './lib/*.ts' })
    expect(value.content).toMatchSnapshot()
    expect(value.files).toMatchSnapshot()
    expect(alignForSnap(value.out)).toContain(`libs/utils/index.js`)
  })

  it('make C undefined ext', async () => {
    // @ts-expect-error import issue
    const { make } = await import('./barrel-maker.cli')
    const { value } = await make({ ext: undefined, target: './{atoms,icons,molecules}/*.tsx' })
    expect(value.content).toMatchSnapshot()
    expect(value.files).toMatchSnapshot()
    expect(alignForSnap(value.out)).toContain(`libs/utils/index.ts`)
  })

  it('main A empty args', async () => {
    // @ts-expect-error import issue
    const { main } = await import('./barrel-maker.cli')
    const { error } = Result.unwrap(await main(['node', 'script.ts']))
    expect(error).toMatchInlineSnapshot(`"missing target argument"`)
  })

  it('main B with args', async () => {
    // @ts-expect-error import issue
    const { main } = await import('./barrel-maker.cli')
    const { value } = Result.unwrap(await main(['node', 'script.ts', '--target=./{atoms,icons,molecules}/*.tsx', '--ext=.js', '--index=index.js']))
    // @ts-expect-error import issue
    expect(value?.content).toMatchSnapshot()
    // @ts-expect-error import issue
    expect(value?.files).toMatchSnapshot()
    // @ts-expect-error import issue
    expect(alignForSnap(value?.out)).toContain(`libs/utils/index.js`)
  })
})
