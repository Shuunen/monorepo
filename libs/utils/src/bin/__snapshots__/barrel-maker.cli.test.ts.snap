// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`barrel-maker cli > main B with args 1`] = `
"export * from './atoms/Button.js'
export * from './icons/Icon.js'
export * from './molecules/Card.js'
export type * from './atoms/types.js'
"
`;

exports[`barrel-maker cli > main B with args 2`] = `
[
  "atoms/Button.tsx",
  "atoms/types.ts",
  "icons/Icon.tsx",
  "molecules/Card.tsx",
  "molecules/Card.stories.tsx",
]
`;

exports[`barrel-maker cli > make A ext empty and default index.ts 1`] = `
"export * from './atoms/Button'
export * from './icons/Icon'
export * from './molecules/Card'
export type * from './atoms/types'
"
`;

exports[`barrel-maker cli > make A ext empty and default index.ts 2`] = `
[
  "atoms/Button.tsx",
  "atoms/types.ts",
  "icons/Icon.tsx",
  "molecules/Card.tsx",
  "molecules/Card.stories.tsx",
]
`;

exports[`barrel-maker cli > make B index.js and ext .js 1`] = `
"export * from './atoms/Button.js'
export * from './icons/Icon.js'
export * from './molecules/Card.js'
export * from './molecules/Card.js'
export type * from './atoms/types.js'
"
`;

exports[`barrel-maker cli > make B index.js and ext .js 2`] = `
[
  "atoms/Button.tsx",
  "atoms/types.ts",
  "icons/Icon.tsx",
  "molecules/Card.tsx",
  "molecules/Card.stories.tsx",
]
`;

exports[`barrel-maker cli > make C undefined ext 1`] = `
"export * from './atoms/Button.tsx'
export * from './icons/Icon.tsx'
export * from './molecules/Card.tsx'
export type * from './atoms/types.ts'
"
`;

exports[`barrel-maker cli > make C undefined ext 2`] = `
[
  "atoms/Button.tsx",
  "atoms/types.ts",
  "icons/Icon.tsx",
  "molecules/Card.tsx",
  "molecules/Card.stories.tsx",
]
`;
